benchmark/syclblas/blas2/spmv.cpp

```diff
@@ -0,0 +1,146 @@
+  std::vector<scalar_t> v_x =
+      blas_benchmark::utils::random_data<scalar_t>(xlen);
+  std::vector<scalar_t> v_y =
+      blas_benchmark::utils::random_data<scalar_t>(ylen);
+
+  auto m_a_gpu = blas::make_sycl_iterator_buffer<scalar_t>(m_a, m_a.size());
+  auto v_x_gpu = blas::make_sycl_iterator_buffer<scalar_t>(v_x, xlen);
+  auto v_y_gpu = blas::make_sycl_iterator_buffer<scalar_t>(v_y, ylen);
+
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  std::vector<scalar_t> v_y_ref = v_y;
+  reference_blas::spmv(uplo_str, n, alpha, m_a.data(), v_x.data(), incX, beta,
+                       v_y_ref.data(), incY);
+  std::vector<scalar_t> v_y_temp = v_y;
+  {
+    auto v_y_temp_gpu =
+        blas::make_sycl_iterator_buffer<scalar_t>(v_y_temp, ylen);
+    auto event = _spmv(sb_handle, *uplo_str, n, alpha, m_a_gpu, v_x_gpu, incX,
+                       beta, v_y_temp_gpu, incY);
+    sb_handle.wait();
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors(v_y_temp, v_y_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif
+
+  auto blas_method_def = [&]() -> std::vector<cl::sycl::event> {
+    auto event = _spmv(sb_handle, *uplo_str, n, alpha, m_a_gpu, v_x_gpu, incX,
+                       beta, v_y_gpu, incY);
+    sb_handle.wait(event);
+    return event;
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_method_def);
+  sb_handle.wait();
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
+  state.SetItemsProcessed(state.iterations() * state.counters["n_fl_ops"]);
+  state.SetBytesProcessed(state.iterations() *
+                          state.counters["bytes_processed"]);
+
+  blas_benchmark::utils::calc_avg_counters(state);
+}
+
+template <typename scalar_t>
+void register_benchmark(blas_benchmark::Args& args,
+                        blas::SB_Handle* sb_handle_ptr, bool* success) {
+  // spmv and  symv use the same set of params, so reuse the symv function
+  auto spmv_params = blas_benchmark::utils::get_symv_params<scalar_t>(args);
+
+  for (auto p : spmv_params) {
+    std::string uplos;
+    index_t n;
+    scalar_t alpha, beta;
+    std::tie(uplos, n, alpha, beta) = p;
+
+    auto BM_lambda = [&](benchmark::State& st, blas::SB_Handle* sb_handle_ptr,
+                         std::string uplos, index_t n, scalar_t alpha,
+                         scalar_t beta, bool* success) {
+      run<scalar_t>(st, sb_handle_ptr, uplos, n, alpha, beta, success);
+    };
+    benchmark::RegisterBenchmark(
+        get_name<scalar_t>(uplos, n, alpha, beta).c_str(), BM_lambda,
+        sb_handle_ptr, uplos, n, alpha, beta, success);
```

Comment:
Thank you @pgorlani.
Missing `->UseRealTime()` when register the benchmark, can you add it?
```suggestion
        sb_handle_ptr, uplos, n, alpha, beta, success)
        ->UseRealTime();
```

Reply:
Thanks for your review, @s-Nick.

Done.